lib/connect.c

```diff
@@ -694,7 +694,7 @@ static CURLcode is_connected(struct Curl_cfilter *cf,
   for(i = 0; i < sizeof(ctx->baller)/sizeof(ctx->baller[0]); i++) {
     struct eyeballer *baller = ctx->baller[i];
     CURL_TRC_CF(data, cf, "%s assess started=%d, result=%d",
-                baller?baller->name:NULL,
+                baller?baller->name:"?",
```

Comment:
I would abstain from including this dozen line warning in the commit message. It's simply that printf-like functions %s format specifier have undefined behavior for NULL. ([technically](https://stackoverflow.com/questions/11589342/what-is-the-behavior-of-printing-null-with-printfs-s-specifier))

All the other loops do an if(!baller) continue; so I have an alternate proposal

~~~diff
diff --git a/lib/connect.c b/lib/connect.c
index c7ba3e2..92eebf0 100644
--- a/lib/connect.c
+++ b/lib/connect.c
@@ -693,11 +693,11 @@ evaluate:
   result = CURLE_COULDNT_CONNECT;
   for(i = 0; i < sizeof(ctx->baller)/sizeof(ctx->baller[0]); i++) {
     struct eyeballer *baller = ctx->baller[i];
+    if(!baller)
+      continue;
     CURL_TRC_CF(data, cf, "%s assess started=%d, result=%d",
-                baller?baller->name:NULL,
-                baller?baller->has_started:0,
-                baller?baller->result:0);
-    if(baller && baller->has_started && baller->result) {
+                baller->name, baller->has_started, baller->result);
+    if(baller->has_started && baller->result) {
       result = baller->result;
       break;
     }
~~~

Ref: #10284

Reply:
You're right. I'll amend the patch too.

I'm seeing many warnings and issues (not necessarily code or even curl-specific) that I stuck to the habit of keeping a copy and a log reference to keep track of them and look them up later if necessary. Surely for this case it will be enough to keep it in the PR, not in the commit.